src/main/java/org/opentripplanner/routing/graph/GraphIndex.java

```diff
@@ -441,7 +442,7 @@ protected boolean lessThan(TripTimeShort tripTimeShort, TripTimeShort t1) {
                 int sidx = 0;
                 for (Stop currStop : pattern.stopPattern.stops) {
                     if (currStop == stop) {
-                        if(pattern.stopPattern.pickups[sidx] == pattern.stopPattern.PICKDROP_NONE) continue;
+                        if(omitNonPickups && pattern.stopPattern.pickups[sidx] == pattern.stopPattern.PICKDROP_NONE) continue;
```

Comment:
I wonder if this will generally exclude the final stop in a trip, and whether that is what you want.